File: java/socketd/src/main/java/org/noear/socketd/exception/SocketdAlarmException.java
package org.noear.socketd.exception;

import org.noear.socketd.transport.core.Message;

/**
 * 告警异常
 *
 * @author noear
 * @since 2.0
 */
public class SocketdAlarmException extends SocketdException {
    private Message from;

    /**
     * 获取来源消息
     */
    public Message getFrom() {
        return from;
    }

    public SocketdAlarmException(Message from) {
        super(from.dataAsString());
        this.from = from;
    }
}

File: dotnet/socketd/org.noear.socketd/exception/SocketdAlarmException.cs
using org.noear.socketd.transport.core;

namespace org.noear.socketd.exception;


/**
 * 告警异常
 *
 * @author noear
 * @since 2.0
 */
public class SocketdAlarmException : SocketdException
{
    private IMessage alarm;

    /**
     * 获取告警
     */
    public IMessage getAlarm()
    {
        return alarm;
    }

    public SocketdAlarmException(IMessage alarm) : base(alarm.dataAsString())
    {
        this.alarm = alarm;
    }
}